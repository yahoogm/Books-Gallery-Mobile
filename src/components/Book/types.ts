import {VolumeInfo} from '../../types/types';

type DetailBookType = {
  volumeInfo: VolumeInfo;
};

export type BookProps = {
  uriDetailBook: string;
  authorsName: string[] | string;
  categories: string[] | string;
  detailBook: DetailBookType;
  setShowBook: React.Dispatch<React.SetStateAction<boolean>>;
  refReadBook: React.RefObject<any>;
};

export type BookContainerProps = {
  setShowBook: React.Dispatch<React.SetStateAction<boolean>>;
  refReadBook: React.RefObject<any>;
};
