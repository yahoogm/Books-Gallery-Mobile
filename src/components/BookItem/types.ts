export type BookItemProps = {
  detailBook: string | number | string[];
  label: string;
};
