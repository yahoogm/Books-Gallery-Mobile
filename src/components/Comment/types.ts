import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {BookItem, Comment, RootStackParamList} from '../../types/types';
import {ObjectSchema} from 'yup';

export type FormCommentValues = {
  comment: string;
};

export type AddCommentValues = {
  userName: string;
  bookId: string;
  profilePic: string;
  id: string;
  ulasan: string;
  userId: string;
  createdAt: string;
  updatedAt: string;
};

// type for handling navigation to login page
export type NavigationProp = NativeStackNavigationProp<
  RootStackParamList,
  'Login'
>;

export type ToastRenderProps = {
  id: string;
};

export type UserItem = {
  email: string;
  familyName: string;
  givenName: string;
  id: string;
  name: string;
  photo: string;
};

export type HandleBookCommentFunction = {
  handleNavigateUserNotLogin: () => void;
  handleAddCommentBook: (review: AddCommentValues) => void;
  handleDeleteCommentBook: () => void;
  handleUpdateCommentBook: (ulasan: string, updateAt: string) => void;
  getIdDocumentFromFirestore: (id: string) => void;
};

export type CommentProps = {
  refOpenDeleteModal: React.RefObject<any>;
  deleteModal: boolean;
  setDeleteModal: React.Dispatch<React.SetStateAction<boolean>>;
  handleButton: HandleBookCommentFunction;
  isEdit: boolean;
  setIsEdit: React.Dispatch<React.SetStateAction<boolean>>;
  commentSchema: ObjectSchema<FormCommentValues>;
  initialValues: FormCommentValues;
  isLogin: boolean;
  user: UserItem;
  detailBook: BookItem;
  commentBook: Array<Comment>;
};
