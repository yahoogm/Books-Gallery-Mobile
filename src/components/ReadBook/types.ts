export type ReadBookProps = {
  setShowBook: React.Dispatch<React.SetStateAction<boolean>>;
  showBook: boolean;
  refReadBook: React.RefObject<any>;
  isbn: number;
};
