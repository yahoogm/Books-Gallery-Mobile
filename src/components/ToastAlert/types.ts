export type ToastAlertProps = {
  id: string;
  message: string;
  action: 'success' | 'info' | 'error' | 'warning' | 'attention';
};
