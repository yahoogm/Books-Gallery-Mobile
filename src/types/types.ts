type VolumeInfo = {
  title: string;
  publishedDate: string;
  description: string;
  authors: Array<string>;
  publisher: string;
  pageCount: number;
  imageLinks: {smallThumbnail: string; thumbnail: string; large?: string};
  categories: Array<string>;
  industryIdentifiers: Array<{identifier: string; type: string}>;
};

export type Review = {
  bookId: string;
  createdAt: {};
  id: string;
  profilePic: string;
  ulasan: string;
  updatedAt: {};
  userId: string;
  userName: string;
};

export type BookItem = {
  id: string;
  volumeInfo: VolumeInfo;
};

export type BookState = {
  type: string;
  reviewId: string;
  detailBook: BookItem;
  books: {
    items: [];
  };
  reviewBook: Array<Review>;
};

export type userItem = {
  email: string;
  familyName: string;
  givenName: string;
  id: string;
  name: string;
  photo: string;
};

export type RootStackParamList = {
  DetailBook: {bookId: string};
  Login: undefined;
};
