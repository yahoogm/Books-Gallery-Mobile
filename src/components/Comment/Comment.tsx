import {
  Button,
  ButtonText,
  Heading,
  Text,
  View,
  VStack,
  AvatarFallbackText,
  Box,
  Avatar,
  AvatarImage,
  ModalBackdrop,
  ModalContent,
  Icon,
  Modal,
  Textarea,
  TextareaInput,
  GripVerticalIcon,
  Menu,
  MenuItem,
  MenuItemLabel,
  EditIcon,
  TrashIcon,
  ButtonIcon,
  ModalFooter,
  ModalHeader,
} from '@gluestack-ui/themed';
import {Formik} from 'formik';
import {v4 as uuidv4} from 'uuid';
import {CommentProps} from './types';

const Comment: React.FC<CommentProps> = ({
  refOpenDeleteModal,
  deleteModal,
  setDeleteModal,
  handleButton,
  isEdit,
  setIsEdit,
  commentSchema,
  initialValues,
  isLogin,
  user,
  detailBook,
  commentBook,
}) => {
  const {
    handleNavigateUserNotLogin,
    handleAddCommentBook,
    handleUpdateCommentBook,
    handleDeleteCommentBook,
    getIdDocumentFromFirestore,
  } = handleButton;
  return (
    <View>
      <Heading>Komentar</Heading>
      <Formik
        validationSchema={commentSchema}
        initialValues={initialValues}
        onSubmit={(values, {resetForm}) => {
          const date = new Date();
          const id = uuidv4();

          isEdit
            ? handleUpdateCommentBook(values.comment, date.toISOString())
            : handleAddCommentBook({
                userName: user.name,
                bookId: detailBook.id,
                profilePic: user.photo,
                id: id,
                ulasan: values.comment,
                userId: user.id,
                createdAt: date.toISOString(),
                updatedAt: date.toISOString(),
              });

          resetForm();
        }}>
        {({handleChange, handleSubmit, values, setFieldValue}) => (
          <View>
            <Textarea
              size="sm"
              isRequired={true}
              isReadOnly={false}
              isInvalid={false}
              isDisabled={false}>
              <TextareaInput
                placeholder="Ketikkan komentar anda..."
                onChangeText={handleChange('comment')}
                value={values.comment}
                fontSize={'$sm'}
                maxLength={200}
              />
            </Textarea>

            <View display="flex" flexDirection="row" gap={4}>
              <Button
                size="sm"
                width={'$1/4'}
                action="positive"
                marginVertical={10}
                isDisabled={values.comment.length < 10 ? true : false}
                onPress={() =>
                  isLogin ? handleSubmit() : handleNavigateUserNotLogin()
                }>
                <ButtonText>{isEdit ? 'Edit' : 'Kirim'}</ButtonText>
              </Button>

              {isEdit && (
                <Button
                  size="sm"
                  width={'$1/4'}
                  action="positive"
                  marginVertical={10}
                  onPress={() => {
                    setIsEdit(false);
                    setFieldValue('comment', '');
                  }}>
                  <ButtonText>Batal</ButtonText>
                </Button>
              )}
            </View>

            {commentBook.length !== 0 ? (
              commentBook.map(book => {
                return (
                  <Box display="flex" flexDirection="row" mt={10} key={book.id}>
                    <Avatar mr="$3">
                      <AvatarFallbackText fontFamily="$heading">
                        Picture
                      </AvatarFallbackText>
                      <AvatarImage
                        alt="gambar"
                        source={{
                          uri: book.profilePic
                            ? book.profilePic
                            : 'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8dXNlcnxlbnwwfHwwfHw%3D&auto=format&fit=crop&w=800&q=60',
                        }}
                      />
                    </Avatar>
                    <VStack flex={1}>
                      <Box display="flex">
                        <Heading size="sm" fontFamily="$heading" mb={1}>
                          {book.userName}
                        </Heading>
                        <Text size="xs">{book.createdAt}</Text>
                      </Box>

                      <Text size="sm" color="black">
                        {book.ulasan}
                      </Text>
                    </VStack>

                    <VStack>
                      <Menu
                        placement="left bottom"
                        backgroundColor="$coolGray200"
                        width={'$full'}
                        margin={0}
                        padding={0}
                        trigger={({...triggerProps}) => {
                          return (
                            <Button
                              {...triggerProps}
                              w="$1"
                              backgroundColor="white">
                              <ButtonIcon as={GripVerticalIcon} color="black" />
                            </Button>
                          );
                        }}>
                        <MenuItem
                          key="Edit"
                          textValue="Edit"
                          onPress={() => {
                            getIdDocumentFromFirestore(book.id);
                            setFieldValue('review', book.ulasan);
                            setIsEdit(true);
                          }}>
                          <Icon as={EditIcon} mr={'$2'} color="green" />
                          <MenuItemLabel color="green">Edit</MenuItemLabel>
                        </MenuItem>

                        <MenuItem
                          key="Hapus"
                          textValue="Hapus"
                          onPress={() => {
                            setDeleteModal(true);
                            getIdDocumentFromFirestore(book.id);
                          }}
                          ref={refOpenDeleteModal}>
                          <Icon as={TrashIcon} mr={'$2'} color="red" />
                          <MenuItemLabel color="red">Hapus</MenuItemLabel>
                        </MenuItem>
                      </Menu>

                      <Modal
                        isOpen={deleteModal}
                        onClose={() => {
                          setDeleteModal(false);
                        }}
                        finalFocusRef={refOpenDeleteModal}>
                        <ModalBackdrop />
                        <ModalContent>
                          <ModalHeader>
                            <Heading size="lg">
                              Yakin ingin menghapus komentar?
                            </Heading>
                          </ModalHeader>
                          <ModalFooter>
                            <Button
                              variant="outline"
                              size="sm"
                              action="secondary"
                              mr="$3"
                              onPress={() => {
                                setDeleteModal(false);
                              }}>
                              <ButtonText>Batal</ButtonText>
                            </Button>
                            <Button
                              size="sm"
                              action="negative"
                              borderWidth="$0"
                              onPress={() => {
                                setDeleteModal(false);
                                handleDeleteCommentBook();
                              }}>
                              <ButtonText>Hapus</ButtonText>
                            </Button>
                          </ModalFooter>
                        </ModalContent>
                      </Modal>
                    </VStack>
                  </Box>
                );
              })
            ) : (
              <Text>Tidak ada komentar</Text>
            )}
          </View>
        )}
      </Formik>
    </View>
  );
};

export default Comment;
