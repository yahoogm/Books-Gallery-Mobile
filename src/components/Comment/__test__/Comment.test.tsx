import {act, fireEvent, render} from '@testing-library/react-native';
import {CommentProps} from '../types';
import Comment from '../Comment';
import {createRef} from 'react';
import {GluestackUIProvider} from '@gluestack-ui/themed';
import {config} from '@gluestack-ui/config';

describe('Comment', () => {
  const refOpenDeleteModal = createRef<any>();
  const mockSetDeleteModal = jest.fn();
  const mockHandleButton = {
    handleNavigateUserNotLogin: jest.fn(),
    handleAddCommentBook: jest.fn(),
    handleDeleteCommentBook: jest.fn(),
    handleUpdateCommentBook: jest.fn(),
    getIdDocumentFromFirestore: jest.fn(),
    handleSubmit: jest.fn(),
  };
  const mockSetIsEdit = jest.fn();

  const renderComponent = (props: CommentProps) =>
    render(
      <GluestackUIProvider config={config}>
        <Comment {...props} />
      </GluestackUIProvider>,
    );

  it('match snapshot when renders and given props', () => {
    const userItem = {
      email: 'testing@gmail.com',
      familyName: 'test',
      givenName: 'ing',
      id: '12345',
      name: 'testing',
      photo: 'https://google.com',
    };
    const volumenInfo = {
      title: 'testing',
      publishedDate: 'testing',
      description: 'testing',
      authors: ['testing', 'testing'],
      publisher: 'testing',
      pageCount: 100,
      imageLinks: {
        smallThumbnail: 'https://google.com',
        thumbnail: 'https://google.com',
        large: 'https://google.com',
      },
      categories: ['testing', 'testing'],
      industryIdentifiers: [{identifier: 'testing', type: 'testing'}],
    };
    const bookItem = {
      id: '123456',
      volumeInfo: volumenInfo,
    };

    const comment = {
      bookId: 'testing',
      createdAt: 'testing',
      id: 'testing',
      profilePic: 'https://google.com',
      ulasan: 'testing',
      updatedAt: 'testing',
      userId: 'testing',
      userName: 'testing',
    };

    const props = {
      refOpenDeleteModal: refOpenDeleteModal,
      deleteModal: false,
      setDeleteModal: mockSetDeleteModal,
      handleButton: mockHandleButton,
      isEdit: false,
      setIsEdit: mockSetIsEdit,
      initialValues: {comment: 'berhasil'},
      isLogin: false,
      user: userItem,
      detailBook: bookItem,
      commentBook: [comment],
    };

    const tree = renderComponent(props);
    const instance = tree.toJSON();
    expect(instance).toMatchSnapshot();
  });

  it('renders correctly if comment value same with props', async () => {
    const userItem = {
      email: 'testing@gmail.com',
      familyName: 'test',
      givenName: 'ing',
      id: '12345',
      name: 'testing',
      photo: 'https://google.com',
    };
    const volumenInfo = {
      title: 'testing',
      publishedDate: 'testing',
      description: 'testing',
      authors: ['testing', 'testing'],
      publisher: 'testing',
      pageCount: 100,
      imageLinks: {
        smallThumbnail: 'https://google.com',
        thumbnail: 'https://google.com',
        large: 'https://google.com',
      },
      categories: ['testing', 'testing'],
      industryIdentifiers: [{identifier: 'testing', type: 'testing'}],
    };
    const bookItem = {
      id: '123456',
      volumeInfo: volumenInfo,
    };

    const comment = {
      bookId: 'testing',
      createdAt: 'testing',
      id: 'testing',
      profilePic: 'https://google.com',
      ulasan: 'testing',
      updatedAt: 'testing',
      userId: 'testing',
      userName: 'testing',
    };

    const props = {
      refOpenDeleteModal: refOpenDeleteModal,
      deleteModal: false,
      setDeleteModal: mockSetDeleteModal,
      handleButton: mockHandleButton,
      isEdit: false,
      setIsEdit: mockSetIsEdit,
      initialValues: {comment: 'testing'},
      isLogin: false,
      user: userItem,
      detailBook: bookItem,
      commentBook: [comment],
    };

    const tree = renderComponent(props);
    const instance = tree.getByTestId('testCommentTextArea');

    await act(async () => {
      fireEvent.changeText(instance, 'testing');
    });

    expect(instance.props.value).toBe('testing');
  });

  it('should disable submit button if comment length is less than 10 characters', async () => {
    const userItem = {
      email: 'testing@gmail.com',
      familyName: 'test',
      givenName: 'ing',
      id: '12345',
      name: 'testing',
      photo: 'https://google.com',
    };
    const volumenInfo = {
      title: 'testing',
      publishedDate: 'testing',
      description: 'testing',
      authors: ['testing', 'testing'],
      publisher: 'testing',
      pageCount: 100,
      imageLinks: {
        smallThumbnail: 'https://google.com',
        thumbnail: 'https://google.com',
        large: 'https://google.com',
      },
      categories: ['testing', 'testing'],
      industryIdentifiers: [{identifier: 'testing', type: 'testing'}],
    };
    const bookItem = {
      id: '123456',
      volumeInfo: volumenInfo,
    };

    const comment = {
      bookId: 'testing',
      createdAt: 'testing',
      id: 'testing',
      profilePic: 'https://google.com',
      ulasan: 'testing',
      updatedAt: 'testing',
      userId: 'testing',
      userName: 'testing',
    };

    const props = {
      refOpenDeleteModal: refOpenDeleteModal,
      deleteModal: false,
      setDeleteModal: mockSetDeleteModal,
      handleButton: mockHandleButton,
      isEdit: false,
      setIsEdit: mockSetIsEdit,
      initialValues: {comment: 'testing'},
      isLogin: false,
      user: userItem,
      detailBook: bookItem,
      commentBook: [comment],
    };

    const tree = renderComponent(props);
    const input = tree.getByTestId('testCommentTextArea');
    const button = tree.getByTestId('testButtonSubmit');

    expect(button.props.states.disabled).toBe(true);

    await act(async () => fireEvent.changeText(input, 'testing'));

    expect(button.props.states.disabled).toBe(true);

    await act(async () =>
      fireEvent.changeText(input, 'testing to make button isDisable false'),
    );

    expect(button.props.states.disabled).toBe(false);
  });

  it('should call handleSubmit when the button is pressed and isLogin true', async () => {
    const userItem = {
      email: 'testing@gmail.com',
      familyName: 'test',
      givenName: 'ing',
      id: '12345',
      name: 'testing',
      photo: 'https://google.com',
    };
    const volumenInfo = {
      title: 'testing',
      publishedDate: 'testing',
      description: 'testing',
      authors: ['testing', 'testing'],
      publisher: 'testing',
      pageCount: 100,
      imageLinks: {
        smallThumbnail: 'https://google.com',
        thumbnail: 'https://google.com',
        large: 'https://google.com',
      },
      categories: ['testing', 'testing'],
      industryIdentifiers: [{identifier: 'testing', type: 'testing'}],
    };
    const bookItem = {
      id: '123456',
      volumeInfo: volumenInfo,
    };

    const comment = {
      bookId: 'testing',
      createdAt: 'testing',
      id: 'testing',
      profilePic: 'https://google.com',
      ulasan: 'testing',
      updatedAt: 'testing',
      userId: 'testing',
      userName: 'testing',
    };

    const props = {
      refOpenDeleteModal: refOpenDeleteModal,
      deleteModal: false,
      setDeleteModal: mockSetDeleteModal,
      handleButton: mockHandleButton,
      isEdit: false,
      setIsEdit: mockSetIsEdit,
      initialValues: {comment: 'testing'},
      isLogin: false,
      user: userItem,
      detailBook: bookItem,
      commentBook: [comment],
    };

    const tree = renderComponent(props);
    const input = tree.getByTestId('testCommentTextArea');
    const button = tree.getByTestId('testButtonSubmit');

    await act(() =>
      fireEvent.changeText(input, 'testing for handle submit button pressed'),
    );
    expect(button.props.states.disabled).toBe(false);

    await act(() => fireEvent.press(button));
    expect(mockHandleButton.handleNavigateUserNotLogin).toHaveBeenCalled();
  });

  it('should render button text kirim if isEdit props false', async () => {
    const userItem = {
      email: 'testing@gmail.com',
      familyName: 'test',
      givenName: 'ing',
      id: '12345',
      name: 'testing',
      photo: 'https://google.com',
    };
    const volumenInfo = {
      title: 'testing',
      publishedDate: 'testing',
      description: 'testing',
      authors: ['testing', 'testing'],
      publisher: 'testing',
      pageCount: 100,
      imageLinks: {
        smallThumbnail: 'https://google.com',
        thumbnail: 'https://google.com',
        large: 'https://google.com',
      },
      categories: ['testing', 'testing'],
      industryIdentifiers: [{identifier: 'testing', type: 'testing'}],
    };
    const bookItem = {
      id: '123456',
      volumeInfo: volumenInfo,
    };

    const comment = {
      bookId: 'testing',
      createdAt: 'testing',
      id: 'testing',
      profilePic: 'https://google.com',
      ulasan: 'testing',
      updatedAt: 'testing',
      userId: 'testing',
      userName: 'testing',
    };

    const props = {
      refOpenDeleteModal: refOpenDeleteModal,
      deleteModal: false,
      setDeleteModal: mockSetDeleteModal,
      handleButton: mockHandleButton,
      isEdit: false,
      setIsEdit: mockSetIsEdit,
      initialValues: {comment: 'testing'},
      isLogin: false,
      user: userItem,
      detailBook: bookItem,
      commentBook: [comment],
    };

    const tree = renderComponent(props);
    const button = tree.getByTestId('testButtonTextSubmit');

    expect(button.props.children).toBe('Kirim');
  });

  it('should render button text edit if isEdit props true', async () => {
    const userItem = {
      email: 'testing@gmail.com',
      familyName: 'test',
      givenName: 'ing',
      id: '12345',
      name: 'testing',
      photo: 'https://google.com',
    };
    const volumenInfo = {
      title: 'testing',
      publishedDate: 'testing',
      description: 'testing',
      authors: ['testing', 'testing'],
      publisher: 'testing',
      pageCount: 100,
      imageLinks: {
        smallThumbnail: 'https://google.com',
        thumbnail: 'https://google.com',
        large: 'https://google.com',
      },
      categories: ['testing', 'testing'],
      industryIdentifiers: [{identifier: 'testing', type: 'testing'}],
    };
    const bookItem = {
      id: '123456',
      volumeInfo: volumenInfo,
    };

    const comment = {
      bookId: 'testing',
      createdAt: 'testing',
      id: 'testing',
      profilePic: 'https://google.com',
      ulasan: 'testing',
      updatedAt: 'testing',
      userId: 'testing',
      userName: 'testing',
    };

    const props = {
      refOpenDeleteModal: refOpenDeleteModal,
      deleteModal: false,
      setDeleteModal: mockSetDeleteModal,
      handleButton: mockHandleButton,
      isEdit: true,
      setIsEdit: mockSetIsEdit,
      initialValues: {comment: 'testing'},
      isLogin: false,
      user: userItem,
      detailBook: bookItem,
      commentBook: [comment],
    };

    const tree = renderComponent(props);
    const button = tree.getByTestId('testButtonTextSubmit');

    expect(button.props.children).toBe('Edit');
  });
});
