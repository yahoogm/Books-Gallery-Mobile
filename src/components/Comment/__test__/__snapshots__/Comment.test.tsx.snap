// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment match snapshot when renders and given props 1`] = `
<View
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  style={
    [
      undefined,
      {},
      undefined,
      {},
      undefined,
      {},
      undefined,
      undefined,
    ]
  }
  sx={{}}
>
  <Text
    accessibilityRole="header"
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    style={
      {
        "color": "#262626",
        "fontFamily": undefined,
        "fontSize": 20,
        "fontWeight": "700",
        "letterSpacing": -0.2,
        "lineHeight": 32,
        "marginVertical": 0,
      }
    }
  >
    Komentar
  </Text>
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    style={
      [
        undefined,
        {},
        undefined,
        {},
        undefined,
        {},
        undefined,
        undefined,
      ]
    }
    sx={{}}
  >
    <View
      data-style=""
      dataSet={
        {
          "disabled": "false",
          "focus": "false",
          "focusVisible": "false",
          "hover": "false",
          "invalid": "false",
          "readonly": "false",
          "required": "true",
          "style": "",
        }
      }
      size="sm"
      states={
        {
          "disabled": undefined,
          "focus": false,
          "focusVisible": false,
          "hover": false,
          "invalid": undefined,
          "readonly": undefined,
          "required": true,
        }
      }
      style={
        [
          undefined,
          {},
          undefined,
          {
            "borderColor": "#D4D4D4",
            "borderRadius": 4,
            "borderWidth": 1,
            "height": 100,
            "width": "100%",
          },
          undefined,
          {},
          undefined,
          {},
          undefined,
          {},
          undefined,
          undefined,
        ]
      }
      sx={{}}
      variant="default"
    >
      <TextInput
        accessible={true}
        aria-label="Input Field"
        aria-required={true}
        aria-selected={false}
        data-style=""
        dataSet={
          {
            "disabled": "false",
            "focus": "false",
            "focusVisible": "false",
            "hover": "false",
            "invalid": "false",
            "readonly": "false",
            "required": "true",
            "style": "",
          }
        }
        editable={true}
        maxLength={200}
        multiline={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onKeyPress={[Function]}
        placeholder="Ketikkan komentar anda..."
        placeholderTextColor="#8C8C8C"
        states={
          {
            "disabled": undefined,
            "focus": false,
            "focusVisible": undefined,
            "hover": false,
            "invalid": undefined,
            "readonly": undefined,
            "required": true,
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
            {
              "color": "#262626",
              "flex": 1,
              "padding": 8,
              "textAlignVertical": "top",
            },
            undefined,
            {},
            undefined,
            {
              "padding": 12,
            },
            undefined,
            {
              "fontSize": 14,
            },
            undefined,
            {},
            undefined,
            {
              "fontSize": 14,
            },
            undefined,
            undefined,
          ]
        }
        sx={
          {
            "fontSize": "$sm",
          }
        }
        testID="testCommentTextArea"
        value="berhasil"
      />
    </View>
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          undefined,
          {},
          undefined,
          {},
          undefined,
          {
            "display": "flex",
            "flexDirection": "row",
            "gap": 4,
          },
          undefined,
          undefined,
        ]
      }
      sx={
        {
          "display": "flex",
          "flexDirection": "row",
          "gap": 4,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        action="positive"
        collapsable={false}
        data-style=""
        dataSet={
          {
            "active": "false",
            "disabled": "true",
            "focus": "false",
            "focusVisible": "false",
            "hover": "false",
            "style": "",
          }
        }
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        size="sm"
        states={
          {
            "active": false,
            "disabled": true,
            "focus": false,
            "focusVisible": false,
            "hover": false,
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
            {
              "alignItems": "center",
              "backgroundColor": "#0077E6",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
            {},
            undefined,
            {
              "backgroundColor": "#348352",
              "borderColor": "#66B584",
            },
            undefined,
            {},
            undefined,
            {
              "height": 36,
              "paddingHorizontal": 16,
            },
            undefined,
            {},
            undefined,
            {
              "opacity": 0.4,
            },
            undefined,
            {
              "backgroundColor": "#348352",
              "borderColor": "#66B584",
            },
            undefined,
            {},
            undefined,
            {
              "height": 36,
              "paddingHorizontal": 16,
            },
            undefined,
            {},
            undefined,
            {
              "marginVertical": 10,
              "width": "25%",
            },
            undefined,
            undefined,
          ]
        }
        sx={
          {
            "marginVertical": 10,
            "width": "$1/4",
          }
        }
        testID="testButtonSubmit"
        variant="solid"
      >
        <Text
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          size="sm"
          style={
            [
              undefined,
              {},
              undefined,
              {
                "color": "#525252",
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              undefined,
              {},
              undefined,
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              undefined,
              undefined,
              {},
              undefined,
              {
                "color": "#FCFCFC",
              },
              undefined,
              {},
              undefined,
              {
                "color": "#FCFCFC",
                "fontWeight": "600",
              },
              undefined,
              {
                "color": "#2A7948",
              },
              undefined,
              {
                "color": "#FCFCFC",
              },
              undefined,
              {},
              undefined,
              {
                "color": "#FCFCFC",
              },
              undefined,
              undefined,
            ]
          }
          sx={{}}
          testID="testButtonTextSubmit"
        >
          Kirim
        </Text>
      </View>
    </View>
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          undefined,
          {},
          undefined,
          {},
          undefined,
          {
            "display": "flex",
            "flexDirection": "row",
            "marginTop": 10,
          },
          undefined,
          undefined,
        ]
      }
      sx={
        {
          "display": "flex",
          "flexDirection": "row",
          "mt": 10,
        }
      }
    >
      <View
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        size="md"
        style={
          [
            undefined,
            {},
            undefined,
            {
              "alignItems": "center",
              "backgroundColor": "#005DB4",
              "borderRadius": 9999,
              "justifyContent": "center",
              "position": "relative",
            },
            undefined,
            {},
            undefined,
            {
              "height": 48,
              "width": 48,
            },
            undefined,
            {
              "height": 48,
              "width": 48,
            },
            undefined,
            {
              "marginRight": 12,
            },
            undefined,
            undefined,
          ]
        }
        sx={
          {
            "mr": "$3",
          }
        }
      >
        <Text
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          size="md"
          style={
            [
              undefined,
              {},
              undefined,
              {
                "color": "#525252",
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              undefined,
              {},
              undefined,
              {
                "fontSize": 16,
              },
              undefined,
              {
                "fontSize": 16,
              },
              undefined,
              {
                "fontFamily": undefined,
              },
              undefined,
              undefined,
              {},
              undefined,
              {
                "color": "#FCFCFC",
                "fontWeight": "600",
                "overflow": "hidden",
                "textTransform": "uppercase",
              },
              undefined,
              {},
              undefined,
              {},
              undefined,
              {
                "fontFamily": undefined,
              },
              undefined,
              undefined,
            ]
          }
          sx={
            {
              "fontFamily": "$heading",
            }
          }
        >
          P
        </Text>
        <Image
          alt="gambar"
          aria-label="gambar"
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          onError={[Function]}
          role="img"
          size="md"
          source={
            {
              "uri": "https://google.com",
            }
          }
          style={
            [
              undefined,
              {},
              undefined,
              {
                "maxWidth": "100%",
              },
              undefined,
              {},
              undefined,
              {
                "height": 80,
                "width": 80,
              },
              undefined,
              undefined,
              {},
              undefined,
              {
                "borderRadius": 9999,
                "height": "100%",
                "position": "absolute",
                "width": "100%",
              },
              undefined,
              {},
              undefined,
              {
                "height": "100%",
                "width": "100%",
              },
              undefined,
              undefined,
            ]
          }
          sx={{}}
        />
      </View>
      <View
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
            {
              "flexDirection": "column",
            },
            undefined,
            {},
            undefined,
            {
              "flex": 1,
            },
            undefined,
            undefined,
          ]
        }
        sx={
          {
            "flex": 1,
          }
        }
      >
        <View
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          style={
            [
              undefined,
              {},
              undefined,
              {},
              undefined,
              {},
              undefined,
              {
                "display": "flex",
              },
              undefined,
              undefined,
            ]
          }
          sx={
            {
              "display": "flex",
            }
          }
        >
          <Text
            accessibilityRole="header"
            data-style=""
            dataSet={
              {
                "style": "",
              }
            }
            style={
              {
                "color": "#262626",
                "fontFamily": undefined,
                "fontSize": 16,
                "fontWeight": "700",
                "letterSpacing": -0.2,
                "lineHeight": 24,
                "marginBottom": 1,
                "marginVertical": 0,
              }
            }
          >
            testing
          </Text>
          <Text
            data-style=""
            dataSet={
              {
                "style": "",
              }
            }
            size="xs"
            style={
              [
                undefined,
                {},
                undefined,
                {
                  "color": "#525252",
                  "fontFamily": undefined,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                undefined,
                {},
                undefined,
                {
                  "fontSize": 12,
                  "lineHeight": 20,
                },
                undefined,
                undefined,
              ]
            }
            sx={{}}
          >
            testing
          </Text>
        </View>
        <Text
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          size="sm"
          style={
            [
              undefined,
              {},
              undefined,
              {
                "color": "#525252",
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              undefined,
              {},
              undefined,
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              undefined,
              {
                "fontSize": 14,
                "lineHeight": 20,
              },
              undefined,
              {
                "color": "black",
              },
              undefined,
              undefined,
            ]
          }
          sx={
            {
              "color": "black",
            }
          }
        >
          testing
        </Text>
      </View>
      <View
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            undefined,
            {
              "flexDirection": "column",
            },
            undefined,
            {},
            undefined,
            undefined,
          ]
        }
        sx={{}}
      />
    </View>
  </View>
</View>
`;
